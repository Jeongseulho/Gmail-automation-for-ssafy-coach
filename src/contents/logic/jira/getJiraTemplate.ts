import { Campus } from '@/types/Campus';
import { Project } from '@/types/Project';
import { Week } from '@/types/Week';
import { formatDate } from '@/utils/formatDate';

export const getJiraTemplate = (cohort: string, date: string, project: Project, campus: Campus, classGroup: string, week: Week, name: string) => {
  const title = `[SSAFY] ${formatDate(date)} ${campus} ${classGroup} ${project} ${week} 주간 Jira 현황 보고서 송부`;
  const content = `
    <div>안녕하십니까, ${cohort} ${campus} 실습코치 ${name}입니다.</div>
    <br>
    <div>${campus} ${classGroup} 금주 주간 Jira 현황 보고서 첨부파일로 송부 드립니다.</div>
    <br>
    <b>특이사항</b>
    <br>
    <div>- 해당 사항 없음.</div>
    <br>
    <div>감사합니다.</div>
    <div>${name} 드림.</div>
  `;
  return {
    title,
    content,
  };
};
