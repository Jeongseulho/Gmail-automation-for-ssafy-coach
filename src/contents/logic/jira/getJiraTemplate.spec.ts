import { getJiraTemplate } from './getJiraTemplate';

describe('getJiraTemplate 함수', () => {
  const { title, content } = getJiraTemplate('10기', '240202', '공통', '구미', '1반', '3주차', '이희진');
  it('제목 템플릿을 반환', () => {
    expect(title).toMatchSnapshot();
  });
  it('내용 템플릿을 반환', () => {
    expect(content).toMatchSnapshot();
  });
});
