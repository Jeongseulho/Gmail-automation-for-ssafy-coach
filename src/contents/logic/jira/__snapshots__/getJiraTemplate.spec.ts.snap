// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getJiraTemplate 함수 내용 템플릿을 반환 1`] = `
"
    <div>안녕하십니까, 10기 구미 실습코치 이희진입니다.</div>
    <br>
    <div>구미 1반 금주 주간 Jira 현황 보고서 첨부파일로 송부 드립니다.</div>
    <br>
    <b>특이사항</b>
    <br>
    <div>- 해당 사항 없음.</div>
    <br>
    <div>감사합니다.</div>
    <div>이희진 드림.</div>
  "
`;

exports[`getJiraTemplate 함수 제목 템플릿을 반환 1`] = `"[SSAFY] 2월 2일(금) 구미 1반 공통 3주차 주간 Jira 현황 보고서 송부"`;
